hack/prefetch-images.sh

```diff
@@ -0,0 +1,57 @@
+set -ex pipefail
+
+source hack/common.sh
+source cluster-up/cluster/$KUBEVIRT_PROVIDER/provider.sh
+source hack/config.sh
+
+function prefetch-images::find_node_names() {
+    if [[ $KUBEVIRT_PROVIDER == "external" ]] || [[ $KUBEVIRT_PROVIDER =~ kind.* ]] || [[ $KUBEVIRT_PROVIDER == "local" ]]; then
+        echo "" # in case of external provider / kind we have no control over the nodes
+    else
+        local nodes=()
+        nodes+=($(_kubectl get nodes -o name | sed "s#node/##g"))
+        echo "${nodes[@]}"
+    fi
+}
+
+# Given a list of images, find nodes, SSH into each node and execute a command to pull image
+function prefetch-images::pull_on_nodes() {
+    local -r containers_to_pull=$@
+    local -r nodes=$(prefetch-images::find_node_names)
+    # only internal providers are supported (we have control over the nodes), and there we know it's docker
+    local -r pull_command="docker"
+    local -r max_retry=10
+
+    for node in ${nodes[@]}; do
+        count=0
+        until ${KUBEVIRT_PATH}cluster-up/ssh.sh ${node} "echo \"${containers_to_pull}\" | xargs \-\-max-args=1 sudo ${pull_command} pull"; do
+            count=$((count + 1))
+            if [ $count -eq max_retry ]; then
+                echo "Failed to '${pull_command} pull' in ${node}" >&2
+                exit 1
+            fi
+            sleep 1
```

Comment:
Is it likely that the command target has recovered after one second? Asked the other way round: might we increase the sleep time with each retry to give it a bit more time in case of repeated failures?

Reply:
I have no idea, I just move this code/extracted common part from cluster-build.sh I suppose someone tested this. Have you seen any problems in this area?

Reply:
but, I think increasing time does not hurt, and might help in the rare situation something takes longer to start/setup